### ISSUE TITLE
Feed and Uploads scroll bar missing in Chrome 

### ISSUE THREAD

**Issue by voschix** [mapillary/mapillary_issues#2230]
- Platform:  `Web`
- System/Browser: `Windows 10 / Chrome version  Version 52.0.2743.116 m`
### Steps to reproduce behavior
1. Open Mapillary.com and log in
2. Open the user side bar
3. Select "Feed" or "Uploads" 
### Expected behavior

There should be a scrolling bar, when there are more feeds/uploads than can be displayed 
### Actual behavior

No scroll bar
### Additional information

The site behaves OK with Firefox or MS Edge on the same computer. The problem seems to be specific to Chrome.

**Comment by gyllen** [comment]
@voschix Does this have anything todo with the size of your browser window?

**Comment by voschix** [comment]
No, it does not dependent on the browser window. I changed the size from minimum to maximum (on a dual monitor arrangement with extended monitor mode)
The missing vertical scroll bar applies to three different sidebars:
1) From the main Mapillary.com page, click the big green "Explore the World", then click the three stripes (between "Mapillary" and the filter icon): the left side menu, hat shows the Leaderboard as top item, has no scroll bar.
2) From the main Mapillary.com page, Sign in,
In the  user sidebar on the right hand side, select "Feed". It has no vertical scroll bar
3) From the main Mapillary.com page, Sign in,
In the  user sidebar on the right hand side, select "Uploads". It has no vertical scroll bar

Everything works fine in Firefox (my standard browser)

In Edge (1) and (2) work fine., but  (3) shows a strange behaviour:
It only shows the oval "Upload"  button, but no uploads. When I sign out (last item in the three-stripes sub-menu), my uploads show up on the right hand side below my user photo and user name, with vertical scroll bar.

**Comment by gyllen** [comment]
@knikel Can you do a quick investigation into this?

**Comment by knikel** [comment]
This is correct behaviour, we want to hide scroll bars where it makes sense and have written specific styles for it. Unfortunately support for that is patchy across browsers, so we have styles that do hide it in all webkit browsers. Doing this in other browsers requires introducing extra offset for the element that is being scrolled and hiding the overflow-y of it. However when the scrollbar is hidden, we should indicate better that the area is scrollable.

**Comment by voschix** [comment]
I would like this to be reopened. 
The contents of the sidebar are not scrollable and there is no scrollbar. 
The problem is specific to Chrome on Windows.

**Comment by pbb72** [comment]
> we want to hide scroll bars where it makes sense

Sorry, but this is utter nonsense. Hiding the scroll bars in an area that is scrollable should **never** make sense. Don't try to be smarter than the rest and think common user interface conventions don't apply to your product.

Scroll bars are **not only** a visual indication of a scrollable area. They are also important, and in some situations vital, tools to perform the scroll. Not everybody has the possibility of using a scroll wheel or a scroll gesture, nor is that the best solution in all situations.

Go study some UI design, before you start hacking away with offsets and overflows to jerk out build-in user interface elements.

**Comment by angelday** [comment]
The way we implemented infinite scrolling in that sidebar breaks the default scroller behaviour (you can check this if you remove the property from the element). We have a couple of ideas how to make the sidebar better (paging, grouping) and we're tracking this issue.

@pbb72 Go easy on @knikel cause he recently started jujitsu and looking for practice partners. ;)

**Comment by pbb72** [comment]
@angelday Sorry, but I have several times had to scroll back in the uploads list several hundreds of items. It's no fun spending half an hour almost continuously flicking your fingers to scroll back. Also tried with a mouse wheel, and it's no fun rolling the scroll wheel for half an hour either. If there had been a scroll bar, then I could have put the mouse cursor at the bottom, and just kept the mouse button down, for automatic scrolling.

**Comment by GITNE** [comment]
Hmm, I have to agree with @pbb72, though I may would have pointed it out a little bit more nicely. :wink: You really should not remove the scroll bar from scrollable areas on purpose (because of indication and usability/accessibility reasons). Alternatively, if you really do not want to have a scrollbar constantly visible—because it is aesthetically more pleasing or because you gain some space—then perhaps you can make it dynamically visible when the user starts scrolling/flinging, similar to what Android does with list views. Yes, a scrollbar is not the sexiest UI element but still useful. IMHO a scrollbar is not only very useful but also convenient, especially if you have to scroll long list quickly. I would call a scroll bar even necessary for long list. However, in any event, long scrolling lists should be avoided in the first place. In this situation grouping should step in.

**Comment by pbb72** [comment]
Sorry for being so blunt 😳. I think the winter is taking it's toll...

**Comment by angelday** [comment]
Keeping it opened as this is a recurring issue. Will move discussion over to the forum.

_mapillary_issues is no longer an active repository._

_Let’s continue discussing features in our forum! We will import this issue as topic to the Mapillary Forum in the coming weeks. We’ll notify here once it’s been done. See you there!_
